### ISSUE TITLE
How to sample from latent distribution

### ISSUE THREAD

**Issue by crlz182** [riannevdberg/sylvester-flows#7]
Hello,

I was wondering how I can generate samples using the decoder network after training. In a VAE, I would just sample from the prior distribution z~N(0,1) and generate a data point using the decoder. In TriangularSylvesterVAE, however, I also have to provide hyperparameters lambda(x) that depend on the input. How can I sample from my latent distribution and generate samples from it?

I am new to normalizing flows in general and would appreciate any help.

**Comment by bfs18** [comment]
I think you can sample from z from N(0, 1) then decode z. The prior distribution is still N(0, 1). The KL in the code is calculated by sampling, because there is closed-form KL when normalizing flow posterior is introduced.
```
    # ln p(z_k)  (not averaged)
    log_p_zk = log_normal_standard(z_k, dim=1)
    # ln q(z_0)  (not averaged)
    log_q_z0 = log_normal_diag(z_0, mean=z_mu, log_var=z_var.log(), dim=1)
    # N E_q0[ ln q(z_0) - ln p(z_k) ]
    summed_logs = torch.sum(log_q_z0 - log_p_zk)

    # sum over batches
    summed_ldj = torch.sum(ldj)

    # ldj = N E_q_z0[\sum_k log |det dz_k/dz_k-1| ]
    kl = (summed_logs - summed_ldj)
```
ln p(z_k) is calculated from N(0, 1). The prior is N(0, 1).

**Comment by bfs18** [comment]
I am also curious about this.

**Comment by bgroenks96** [comment]
Just as in a regular VAE, the latent distribution is conditional on the input. Given an input `x`, you would obtain a `mu` and `sigma` from the encoder, and then draw samples from this latent distribution and feed them through the decoder.

It sounds like what you want is an unconditional prior `p(z)`, which is not possible for a flow-based VAE unless you also define a prior on the flow parameters.
